### PR TITLE
add i18n support for spanish and deutsch language

### PR DESCRIPTION
**Pull request by Ramshell**
# Add gettext

* Every feedback string is wrapped with gettext in order to use the appropriate po file based on the locale.
* Add support for Spanish language.
* Add support for Deutsch language.
* English version will show up in case there is no po file based on the locale config.

### CODE REVIEW

File: po/nudoku.pot [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: nudoku 1.0.0\n"
+"Report-Msgid-Bugs-To: jubalh@iodoru.org\n"
+"POT-Creation-Date: 2018-04-27 19:00-0300\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
```
Comment (jubalh) [inline po/nudoku.pot]: should we not have some value here?

File: src/main.c [hunk 2]
```diff
@@ -141,11 +144,11 @@ static void parse_arguments(int argc, char *argv[])
 				g_provided_stream = strdup(optarg);
 				break;
 			case 'd':
-				if (strcmp(optarg, "easy") == 0)
+				if (strcmp(optarg, _("easy")) == 0)
```
Comment (jubalh) [inline src/main.c]: actually i think the parameters should have the same values in all the languages. so lets use `easy` here and not translate it.

File: src/main.c [hunk 3]
```diff
@@ -141,11 +144,11 @@ static void parse_arguments(int argc, char *argv[])
...
 					g_level = D_EASY;
-				else if (strcmp(optarg, "normal") == 0)
+				else if (strcmp(optarg, _("normal")) == 0)
```
Comment (jubalh) [inline src/main.c]: same as above

File: src/main.c [hunk 4]
```diff
@@ -141,11 +144,11 @@ static void parse_arguments(int argc, char *argv[])
...
 					g_level = D_NORMAL;
-				else if (strcmp(optarg, "hard") == 0)
+				else if (strcmp(optarg, _("hard")) == 0)
```
Comment (jubalh) [inline src/main.c]: same as above

**Review by jubalh — CHANGES_REQUESTED** [review]

File: po/nudoku.pot (hunk 1 above)
Comment (Ramshell) [reply po/nudoku.pot]: hmmm okay, I'll write my email

**Comment by Ramshell** [comment]
Done. We've also added support for Deutsch language ;)

File: po/de.po [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+msgid " S - Solve puzzle\n"
+msgstr " S - Puzzle lösen\n"
+
+#: main.c:281
+msgid " c - Check solution\n"
+msgstr " c - Überprüfen Sie die Lösung\n"
+
+#: main.c:276
+msgid " h - Move left\n"
+msgstr " h - Geh nach links\n"
```
Comment (jubalh) [inline po/de.po]: Lets use the same as for the right side here: Nach links bewegen

File: po/de.po [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+#: main.c:277
+msgid " j - Move down\n"
+msgstr " j - Sich abwärts bewegen\n"
+
+#: main.c:278
+msgid " k - Move up\n"
+msgstr " k - Nach oben gehen\n"
```
Comment (jubalh) [inline po/de.po]: Lets use: Nach unten bewegen

File: po/de.po [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+msgstr " mit Hilfe von %d Tipps"
+
+#: main.c:291
+msgid " x - Delete number\n"
+msgstr " x - Nummer löschen\n"
+
+#: main.c:104
+#, c-format
+msgid "Character %c at position %d is not allowed.\n"
+msgstr "Charakter %c in Position %d ist nicht erlaubt.\n"
```
Comment (jubalh) [inline po/de.po]: Zeichen %c an Position %d ist nicht erlaubt

File: po/de.po [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+msgid "Commands\n"
+msgstr "Befehle\n"
+
+#: main.c:558
+msgid "Correct so far"
+msgstr "Soweit richtig"
+
+#: main.c:513
+msgid "Generating puzzle..."
+msgstr "Generierendes Puzzle..."
```
Comment (jubalh) [inline po/de.po]: Generiere Sudoku

File: po/de.po [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+msgid "Movement\n"
+msgstr "Bewegung\n"
+
+#: main.c:539
+msgid "Not correct"
+msgstr "Nicht richtig"
+
+#: main.c:586
+msgid "Provided hint"
+msgstr "Bereitgestellter Hinweis"
```
Comment (jubalh) [inline po/de.po]: "Tipp gegeben"

**Review by jubalh — CHANGES_REQUESTED** [review]

**Comment by Ramshell** [comment]
Changes done! :+1:

**Comment by jubalh** [comment]
Very nice!
One more thing: It would be good to have this as 3 commits:
 - Add i18n support
 - Add spanish translation
 - Add german translation

Can you do that? (Use `git rebase`, `git commit --amend` etc)

**Comment by Ramshell** [comment]
Hmm should I rebase with other branch (doing those 3 commits) and change the source branch of the pr?

**Comment by jubalh** [comment]
Doesn't matter to me. You can do it in the same branch.
I just would like to have a clean history in the end, so that we can see what was done, and have it nicely seperated per commit.

Maybe copy your directory as a backup in case you mess up and want to try to rebase again :)

**Pull request closed without merging**